mvcc/watchable_store.go

```diff
@@ -167,12 +167,11 @@ func (s *watchableStore) cancelWatcher(wa *watcher) {
 		for _, wb := range s.victims {
 			if wb[wa] != nil {
 				victimBatch = wb
-				break
+				slowWatcherGauge.Dec()
+				delete(victimBatch, wa)
```

Comment:
Can we write an unit test?

Reply:
I can try, @gyuho .

Reply:
@gyuho 
I searched some related tests (under mvcc and integration) but am still looking for example which I can extrapolate.
The test needs to accumulate more than one watcherBatch before the watcher is cancelled.
We need to find a way to assert that the close of watcher cleans all watcherBatch's. Not sure how to do that yet.

Reply:
> Can we write an unit test?

and/or a benchmark test that recreates the conditions leading to the bug this says it is fixing. are there benchmark tests that are release-blocking?

Reply:
I found tools/benchmark/cmd/watch.go where benchMakeWatches bears some resemblance to the tests shown in #11350

Not sure about its role in release process.

Reply:
I searched the output for the 3 TARGET=linux-amd64-integration-x-cpu tests for 
```
	fmt.Printf("Watch creation summary:\n%s", <-rc)
```
from benchMakeWatches().
Didn't find it in output.

Reply:
@gyuho @liggitt 
Kindly suggest the next step.

Reply:
ping @gyuho for review.

Reply:
@tedyu Have you looked into writing an unit test?

Reply:
@gyuho 
Yes I have spent some time from various angles - but haven't come up with new test.

I am still looking for an un-intrusive way of testing this scenario.

Reply:
@gyuho 
Do you have suggestion on how the test can be formulated ?